### ISSUE TITLE
Mac OS Mojave

### ISSUE THREAD

**Issue by bitsbr** [toleda/audio_CloverALC#139]
Will be any support to Mac OS Mojave ? 
Is anyway that I can edit the file to  'force' the script to fun on my Mojave Beta installation ?

**Comment by toleda** [comment]
Yes.
Major issue: Audio ID: 1 and 3 do not work in Mojave, Audio ID: 2 is working
Mojave release will likely have new Audio IDs, backward capability is unknown at this point.

cloverALC, use Audio ID: 2, edit audio_cloverALC...command:
at
# verify system version
case ${gSysVer} in

add
    10.14* ) gSysName="Mojave"
    gSysFolder=kexts/10.14
    gSID=$(csrutil status)
    ;;

**Comment by bitsbr** [comment]
After I made the changes in Audio ID and also the script I receive this output: 

iMac-de-Rafael:Downloads rafael$ sh audio_cloverALC-130.sh  
Agreement
The audio_cloverALC script is for personal use only. Do not distribute
the patch, any or all of the files or the resulting patched AppleHDA.kext
for any reason without permission. The audio_cloverALC script is
provided as is and without any kind of warranty.
 
File: audio_cloverALC-130.command_v0.5
Release Mode
EFI partition is mounted

No audio codec detected
To save a Copy of this Terminal session: Terminal/Shell/Export Text As ...


My audio is ALC887

**Comment by toleda** [comment]
Motherboard?
Attach a copy of IOReg.

**Comment by bitsbr** [comment]
Motherboard: Asus H81M-CS/BR

My IOReg ( I don't know if I made the process right, first time I done this): 
https://ufile.io/c896t

**Comment by toleda** [comment]
File is corrupt.
Use this version: https://github.com/toleda/audio_ALCInjection/blob/master/IORegistryExplorer_v2.1.zip
IORegistryExplorer/File/Save A Copy as ...

**Comment by bitsbr** [comment]
I made another upload with the updated tool,I tested seems to be working now
https://ufile.io/k20xq

**Comment by toleda** [comment]
Installed Voodoo, not supported in this repo.

**Comment by bitsbr** [comment]
@toleda  Yes, when i send  the log i was using vodoo but i made this after tried all steps before when i was not using vodoo before last  IOReg

**Comment by toleda** [comment]
IOReg with Voodoo makes it impossible to see what was wrong before Voodoo. No support with Voodoo installed.  Choices:
a. Continue with Voodoo, issue will be closed
b. As requested previously, attach a useable IOReg without Voodoo.

**Comment by raulmiguelramos** [comment]
Changed:
Add in "audio_cloverALC-130_v0.5.command": 
 >    10.14* ) gSysName="Mojave"
>     gSysFolder=kexts/10.14
>     gSID=$(csrutil status)
>     ;;

After run the modified "audio_cloverALC-130_v0.5.command"

In config.plist
Change: MatchOS from "10.12.x,10.13.x" to "10.12.x,10.13.x,10.14.x" in Kernel and Kext Patches
Change: Audio ID 1 to 2 in devices

Reboot and works on a Gigabyte z97 gaming 5 (Realtek ALC1150 !?)
I did not test with id = 1

Thanks

**Comment by toleda** [comment]
Thanks.

**Comment by raulmiguelramos** [comment]
For troubleshooting propose. I only have Audio if i use the driver UEFI 64 "**OsxAptioFix2Drv-64.efi**" If i use "**OsxAptioFix3Drv-64.efi**".
I use "**OsxAptioFix3Drv-64.efi**" on High Sierra and audio works fine.

**Comment by bitsbr** [comment]
@raulmiguelramos  
I reinstalled Mojave Beta 3 from scratch, and follow your steps editing the config.plist and audio works.
Thank you very much!

**Comment by toleda** [comment]
raulmiguelramos
Try AptioMemoryFix-64.efi, replaces OsxAptioFix2Drv-64.efi and OsxAptioFix3Drv-64.efi

**Comment by raulmiguelramos** [comment]
Was to swear AptioMemoryFix-64.efi was the first one then  OsxAptioFix2Drv-64.efi hibernation support and the late OsxAptioFix3Drv-64.efi with NVRAM support or something in this lines?:
AptioMemoryFix-64.efi and OsxAptioFix2Drv-64.efi works fine in Mojave with this patch. Don't know why OsxAptioFix3Drv-64.efi doesn't work.
I put it here to help someone with the problem.

**Comment by Pro2005** [comment]
Hi guys. Could you help me with my hackintosh
I have Gigabyte B360M D2V motherboard with ALC887 codec
When I try to start audio_cloverALC...command I see
"no audio codec detected" 
Mac OS Mojave Beta 3

https://ufile.io/h2bls
https://ufile.io/f219d
My drivers64UEFI folder contains:
apfs.efi
AppleImageCodec-64.efi
AppleKeyAggregator-64.efi
AppleUITheme-64.efi
AptioMemoryFix-64.efi
DataHubDxe-64.efi
EmuVariableUefi-64.efi
FirmwareVolume-64.efi
FSInject-64.efi
SMCHelper-64.efi
VBoxHfs-64.efi

Any help is appreciated

**Comment by toleda** [comment]
AptioMemoryFix-64.efi and OsxAptioFix2Drv-64.efi works fine in Mojave with this patch: one or the other, not both.

**Comment by toleda** [comment]
Mojave does not support Coffee Lake audio. Coffee Lake audio patch is not working.
Attach Clover boot log.

**Comment by toleda** [comment]
config.plist typo, remove ,
from: 10.13.x,10.14.x, 
to:      10.13.x,10.14.x

![screen shot 2018-07-17 at 10 26 08 pm](https://user-images.githubusercontent.com/3318033/42855932-d5b6a256-8a10-11e8-8e1a-6655da64e7f9.png)

**Comment by Pro2005** [comment]
@toleda done
But the result the same

**Comment by toleda** [comment]
Did this error disappear?
5:969  0:000   - [06]: com.apple.driver.AppleHDAController (10.13-Coffee_Audio_Controller) :: BinPatch :: [OS: 10.14 | MatchOS: 10.13.x,10.14.x, | MatchBuild: All] ==> not allowed

**Comment by gipfelen** [comment]
Your fix works perfectly, but I'm unable to get the microphone working with audio id 2. Any suggestions?

**Comment by gipfelen** [comment]
I fixed the issue by directly using prerelease lilu/audioALC kext files! (now it works with audio-id 1 injected)

**Comment by toleda** [comment]
Apology for the delay in replying.  Still working through Mojave issues.  Will update when Mojave testing is complete.

**Comment by mrPsycho** [comment]
after upgrading to 10.14 from 10.13 sound dissapeared
did that: https://github.com/toleda/audio_CloverALC/issues/139#issuecomment-403296370
and it turned back

thanks @toleda and @raulmiguelramos !

**Comment by bitsbr** [comment]
Guys with this patches to the script my audio was working very well in Mojave with my old GTX 1060, but now, I exchange my GTX 1060 to a Radeon RX 480 8gb sapphire , and I have only HDMI audio. I tried to run the command as always, and put audio id = 2. But now I have only HDMI output .... not green output and pink input anymore. What should I do ?

**Comment by yusabian** [comment]
> Guys with this patches to the script my audio was working very well in Mojave with my old GTX 1060, but now, I exchange my GTX 1060 to a Radeon RX 480 8gb sapphire , and I have only HDMI audio. I tried to run the command as always, and put audio id = 2. But now I have only HDMI output .... not green output and pink input anymore. What should I do ?

Try use AppleALC.kext and Lilu.kext

**Comment by farrvince** [comment]
Hello. Do you have any solution for ALC889 at Mojave? :) I tried with this patch but still no sound.

**Comment by toleda** [comment]
Mojave is not supported, see README. Same 889 files are included with AppleALC.kext/layout=1.
This repo is deprecated.

**Comment by farrvince** [comment]
> Mojave is not supported, see README. Same 889 files are included with AppleALC.kext/layout=1.
> This repo is deprecated.

Thank you. Solved with this: "
10.14* ) gSysName="Mojave"
gSysFolder=kexts/10.14
gSID=$(csrutil status)
;;
"
